### PR TITLE
Damage dealer refactor

### PR DESCRIPTION
**Pull request by dVoorhout**
Refactored everything that had to do with damagedealers, made enemies and obstacles separate things. They both have a spawner which handles everything which has to do with spawning. A spawner can have 1 or more factories. This makes it so when the players get further, the obstacle spawner can switch out static obstacles for dynamic obstacles. Since we decided that enemies and obstacles aren't the same, a general spawner handler wasn't created, they each handle their own spawning.

note: the failing builds was because I forgot to change the tests after the merge

### CODE REVIEW

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java [hunk 1]
```diff
@@ -5,15 +5,16 @@
...
 import com.github.migi_1.Context.model.entity.Camera;
 import com.github.migi_1.Context.model.entity.Carrier;
 import com.github.migi_1.Context.model.entity.CarrierMoveBehaviour;
 import com.github.migi_1.Context.model.entity.Commander;
 import com.github.migi_1.Context.model.entity.Entity;
 import com.github.migi_1.Context.model.entity.EntityMoveBehaviour;
 import com.github.migi_1.Context.model.entity.Platform;
+import com.github.migi_1.Context.obstacle.Obstacle;
+import com.github.migi_1.Context.obstacle.ObstacleSpawner;
+import com.github.migi_1.Context.obstacle.StaticObstacle;
```
Comment (marcelkuipers) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java]: Unused import

File: ContextDesktop/src/main/java/com/github/migi_1/Context/enemy/EnemySpawner.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.github.migi_1.Context.enemy;
+
+
+public class EnemySpawner {
```
Comment (NilsHullegien) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/enemy/EnemySpawner.java:8]: Missing javadoc

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java [hunk 3]
```diff
@@ -189,14 +192,14 @@ private void initSpatials() {
...
             addDisplayable(levelPiece);
         }
-
-        for (DamageDealer damageDealer : damageDealerGenerator.getObstacles()) {
-            addDisplayable(damageDealer);
+        for (Obstacle staticObstacle : obstacleSpawner.getObstacles()) {
+            addDisplayable(staticObstacle);
         }
         for (Path path : levelGenerator.getPathPieces(COMMANDER_LOCATION)) {
             addDisplayable(path);
```
Comment (NilsHullegien) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java:204]: Maybe put these four for-loops in a different method (like an addDisplayables method).

**Comment by dVoorhout** [comment]
will implement comments

**Comment by NilsHullegien** [comment]
Looks good now!

**Comment by marcelkuipers** [comment]
It looks good now.

**Pull request merged**